docs/csharp/language-reference/builtin-types/nullable-value-types.md

```diff
@@ -0,0 +1,131 @@
+As a value type is implicitly convertible to the corresponding nullable value type, you can assign a value to a variable of a nullable value type as you would do that for its underlying value type. You also can assign the `null` value. For example:
+
+[!code-csharp[declare and assign](~/samples/csharp/language-reference/builtin-types/NullableValueTypes.cs#Declaration)]
+
+The default value of a nullable value type represents `null`, that is, it's an instance whose <xref:System.Nullable%601.HasValue%2A?displayProperty=nameWithType> property returns `false`.
+
+## Examination of an instance of a nullable value type
+
+Beginning with C# 7.0, you can use the [`is` operator with a type pattern](../operators/type-testing-and-cast.md#type-testing-with-pattern-matching) to both examine an instance of a nullable value type for `null` and retrieve a value of an underlying type:
+
+[!code-csharp-interactive[use pattern matching](~/samples/csharp/language-reference/builtin-types/NullableValueTypes.cs#PatternMatching)]
+
+You always can use the following read-only properties to examine and get a value of a nullable value type variable:
+
+- <xref:System.Nullable%601.HasValue%2A?displayProperty=nameWithType> indicates whether an instance of a nullable value type has a value of its underlying type.
+
+- <xref:System.Nullable%601.Value%2A?displayProperty=nameWithType> gets the value of an underlying type if <xref:System.Nullable%601.HasValue%2A> is `true`. If <xref:System.Nullable%601.HasValue%2A> is `false`, the <xref:System.Nullable%601.Value%2A> property throws an <xref:System.InvalidOperationException>.
+
+The following example uses the `HasValue` property to test whether the variable contains a value before displaying it:
+
+[!code-csharp-interactive[use HasValue](~/samples/csharp/language-reference/builtin-types/NullableValueTypes.cs#HasValue)]
+
+You also can compare a variable of a nullable value type with `null` instead of using the `HasValue` property, as the following example shows:
+
+[!code-csharp-interactive[use comparison with null](~/samples/csharp/language-reference/builtin-types/NullableValueTypes.cs#CompareWithNull)]
+
+## Conversion from a nullable value type to an underlying type
+
+If you need to assign a value of a nullable value type to a non-nullable value type variable, use the [null-coalescing operator `??`](../operators/null-coalescing-operator.md) to specify the value to be assigned if a value of a nullable value type is `null` (you also can use the <xref:System.Nullable%601.GetValueOrDefault(%600)?displayProperty=nameWithType> method to do that):
+
+[!code-csharp-interactive[?? operator](~/samples/csharp/language-reference/builtin-types/NullableValueTypes.cs#NullCoalescing)]
+
+Use the <xref:System.Nullable%601.GetValueOrDefault?displayProperty=nameWithType> method if the value to be used when a value of a nullable value type is `null` should be the [default](../keywords/default-values-table.md) value of the underlying value type.
```

Comment:
This sentence read very awkward for me. How about this version:

```suggestion
Often, you want to use the [default](../keywords/default-values-table.md) value of the underlying value type in place 
of the `null` of a nullable value type. The <xref:System.Nullable%601.GetValueOrDefault?displayProperty=nameWithType> method accomplishes this.
```

Reply:
@BillWagner I like your wording more. I've edited it a bit and also tried to apply it to the paragraph above.